### PR TITLE
support *.mk files in a replace_in_file function (#502)

### PR DESCRIPTION
**Pull request by slsyy**
These files contains paths, which are used by dependencies. New
sandboxing feature has exposed that omission

Also set configure_in_place=True, so apr-1-config show existing APR_SOURCE_DIR

### CODE REVIEW

File: examples/apache_httpd/BUILD [hunk 1]
```diff
@@ -2,13 +2,15 @@ load("@rules_foreign_cc//tools/build_defs:configure.bzl", "configure_make")
 
 configure_make(
```
Comment (UebelAndre) [inline examples/apache_httpd/BUILD:3]: Given that this was broken but CI continued to produce passing builds, I suspect this is not actually building in CI. Can you check to see that it's hooked up correctly?

**Review by UebelAndre — CHANGES_REQUESTED** [review]

**Comment by UebelAndre** [comment]
Can you talk a bit more about what this function is doing with the files it finds and why including `*.mk` files here makes sense?

File: examples/apache_httpd/BUILD (hunk 1 above)
Comment (slsyy) [reply examples/apache_httpd/BUILD:3]: I don't know. Most of the packages in the `examples` workspace are not used in any way by the CI

Comment (UebelAndre) [reply examples/apache_httpd/BUILD:3]: I think it'd be simple enough to add `apache_httpd` to [this list](https://github.com/bazelbuild/rules_foreign_cc/blob/87df6b25f6c009883da87f07ea680d38780a4d6f/examples/BUILD#L19-L33) to make sure the target is building in CI.

**Comment by slsyy** [comment]
> Can you talk a bit more about what this function is doing with the files it finds and why including `*.mk` files here makes sense?
> 
> (Aside from it fixing a bug in the `apr` example)

I am not sure exactly how it works, because I am not so into autotools.  Without this change `Makefile` generated by `configure` from `:apr_util` try to invoke `:apr` `libutil` from the sandbox, which was used to build `:apr`

```
make[1]: Entering directory '/home/s/.cache/bazel/_bazel_s/574522e525178645dc30ffc4bcb601a7/sandbox/linux-sandbox/11/execroot/rules_foreign_cc_tests/bazel-out/k8-fastbuild/bin/apache_httpd/apr_util.build_tmpdir'
/bin/sh /home/s/.cache/bazel/_bazel_s/574522e525178645dc30ffc4bcb601a7/sandbox/linux-sandbox/10/execroot/rules_foreign_cc_tests/bazel-out/k8-fastbuild/bin/apache_httpd/apr.build_tmpdir/apr/build-1/libtool --silent --mode=compile /usr/bin/gcc -pthread  -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-omit-frame-pointer -fno-canonical-system-headers -Wno-builtin-macro-redefined -D__DATE__="redacted" -D__TIMESTAMP__="redacted" -D__TIME__="redacted" -DHAVE_CONFIG_H  -DLINUX -D_REENTRANT -D_GNU_SOURCE   -I/home/s/.cache/bazel/_bazel_s/574522e525178645dc30ffc4bcb601a7/sandbox/linux-sandbox/11/execroot/rules_foreign_cc_tests/bazel-out/k8-fastbuild/bin/apache_httpd/apr_util.build_tmpdir/include -I/home/s/.cache/bazel/_bazel_s/574522e525178645dc30ffc4bcb601a7/sandbox/linux-sandbox/11/execroot/rules_foreign_cc_tests/bazel-out/k8-fastbuild/bin/apache_httpd/apr_util.build_tmpdir/include/private -I/home/s/.cache/bazel/_bazel_s/574522e525178645dc30ffc4bcb601a7/sandbox/linux-sandbox/11/execroot/rules_foreign_cc_tests/external/apr_util/include/private -I/home/s/.cache/bazel/_bazel_s/574522e525178645dc30ffc4bcb601a7/sandbox/linux-sandbox/11/execroot/rules_foreign_cc_tests/external/apr_util/include  -I/home/s/.cache/bazel/_bazel_s/574522e525178645dc30ffc4bcb601a7/sandbox/linux-sandbox/11/execroot/rules_foreign_cc_tests/bazel-out/k8-fastbuild/bin/apache_httpd/apr_util.ext_build_deps/apr/include/apr-1    -o buckets/apr_brigade.lo -c /home/s/.cache/bazel/_bazel_s/574522e525178645dc30ffc4bcb601a7/sandbox/linux-sandbox/11/execroot/rules_foreign_cc_tests/external/apr_util/buckets/apr_brigade.c && touch buckets/apr_brigade.lo
/bin/sh: /home/s/.cache/bazel/_bazel_s/574522e525178645dc30ffc4bcb601a7/sandbox/linux-sandbox/10/execroot/rules_foreign_cc_tests/bazel-out/k8-fastbuild/bin/apache_httpd/apr.build_tmpdir/apr/build-1/libtool: No such file or directory
```
Notice that the current/`:apr_util` target is using `sandbox/11`,  and the `:apr` target was using `sandbox/10`

File: examples/apache_httpd/BUILD (hunk 1 above)
Comment (slsyy) [reply examples/apache_httpd/BUILD:3]: I will try

Comment (slsyy) [reply examples/apache_httpd/BUILD:3]: I tried on Linux and Mac. Mac fail due to 
```
configure: error: could not determine /private/var/tmp/_bazel_buildkite/8afef7f0076038b0098d9c7983f77386/sandbox/darwin-sandbox/196/execroot/rules_foreign_cc_tests/external/local_config_cc/libtool interface
```
so I have switched to Linux only test suite

**Review by UebelAndre — APPROVED** [review]
Thanks! The only thing outstanding is the CLA. Looks good otherwise 😄

File: examples/apache_httpd/BUILD (hunk 1 above)
Comment (UebelAndre) [reply examples/apache_httpd/BUILD:3]: I think that's fine for the time being. Unfortunate though

**Comment by UebelAndre** [comment]
@slsyy Hi! just wanted to ping about the CLA. There's a change I'd like to get in that would conflict with this one. I don't want to have you unnecessarily redo work as a result 😄

**Comment by slsyy** [comment]
> @slsyy Hi! just wanted to ping about the CLA. There's a change I'd like to get in that would conflict with this one. I don't want to have you unnecessarily redo work as a result 😄

Sorry, I waiting for my company to finish a CLA stuff. Please don't wait for me, because it could take some time

**Comment by UebelAndre** [comment]
This should be solved by https://github.com/bazelbuild/rules_foreign_cc/pull/635

Definitely reopen or create a new issue/PR if it doesn't 😀

**Pull request closed without merging**